htdocs/core/lib/company.lib.php

```diff
@@ -304,17 +304,31 @@ function societe_prepare_head(Societe $object)
 	$head[$h][0] = DOL_URL_ROOT.'/societe/agenda.php?socid='.$object->id;
 	$head[$h][1] = $langs->trans("Events");
 	if (!empty($conf->agenda->enabled) && (!empty($user->rights->agenda->myactions->read) || !empty($user->rights->agenda->allactions->read))) {
+		$nbEvent = 0;
```

Comment:
This may be highly time and cpu consuming for example. So i suggest to add the internal Dolibarr cache feature for this by using :

require_once DOL_DOCUMENT_ROOT.'/core/lib/memory.lib.php';
set_getcache('count_event_'.$object->id, value);
$tmparray = dol_getcache('count_event_'.$object->id);